spec/index.html

```diff
@@ -2881,8 +2898,15 @@ <h3>Insecure Hash Algorithms</h3>
       and implementations of it can be parameterized to use a different
       hash function, without the need to make any changes to the
       canonicalization algorithm itself.
-      However, using a different hash algorithm will generally lead to different results.
+      However, using a different hash algorithm will generally lead to different results;
+      applications making use of this specification should carfully weigh the advantages
+      and disadvantates of using an alternative hash function.
     </p>
+
+    <p class="note">The possible implications of the default hash algorithm
+      becoming insecure are mitegated by that fact that no internal hash
+      values are revealed, and the algorithms are designed to cope
+      with hash collisions.</p>
```

Comment:
It seems to me that "to cope with hash collisions" might be a bit of an overstatement because a hash collision leads to a kind of uncertainty to the output of the algorithm.
As I mentioned in the mailing list, I believe that a hash collision leads to *ties* in for-loops of [step 4](https://www.w3.org/TR/rdf-canon/#ca.4), [step 5](https://www.w3.org/TR/rdf-canon/#ca.5), and [step 5.3](https://www.w3.org/TR/rdf-canon/#ca.5.3) in the Canonicalization Algorithm. These ties can't be deterministically resolved, making the outcome indeterminate. 
To claim it 'copes with collisions', I believe we'd need to incorporate the existence check for ties into the algorithm. However, this might complicate the algorithm and may not be easily accepted so I personally suggest just omitting this phrase here and not to modify the algorithm itself.

Reply:
From what I can tell, there are different kinds of hash collisions:

* Two strings may result in the same hash value.
* Two quads may have the same first-order hash.
* A graph may have homomorphisms (n-degree or poison graph).

My statement is intended to consider  the first case, algorithms such as CRC, MD5, and SHA-1 are more likely to collide on a given input than higher order SHA variants. My understanding that a collision using SHA-256 is already quote low. [One answer on Quara](https://www.quora.com/Is-there-a-collision-in-SHA256#:~:text=Re%20“Is%20there%20a%20collision,256%20%3D%201.16x10%5E77) puts this in 1 chance in 2^256 (presumably with SHA-384 being 1 chance in 2^384). In my view, these are already so low as to be extremely unlikely and effectively of equivalent risk.

The problems @yamdan highlights are in hash 1-degree, and hash n-degree results, which don't relate to the quality of the SHA-256 hash, but in underlying graph homomorphisms which exist independent of any given hash function.

The "copes with collisions" claim is that, in the unlikely event that two distinct quads result in the same hash without having the same string representation, this would promote the problem to the hash n-degrees use case, which could have been avoided.

I think removing a statement entirely doesn't help in conveying the different types of hash collisions involved, and the relative risk of SHA-256 compared with SHA-384.

Reply:
> The "copes with collisions" claim is that, in the unlikely event that two distinct quads result in the same hash without having the same string representation, this would promote the problem to the hash n-degrees use case, which could have been avoided.

Thank you for the explanation. I indeed overlooked that. As you mentioned, in the loop of [step 4](https://www.w3.org/TR/rdf-canon/#ca.4) and [step 5](https://www.w3.org/TR/rdf-canon/#ca.5), we have already anticipated the possibility of collisions with the 1-degree hash and have taken measures to promote it to the Hash N-Degree Quads algorithm. Thus, I believe RDFC-1.0 already *copes with hash 1-degree collisions*.

I am still curious about what would happen if we encounter unexpected n-degree hash collisions in [step 5.3](https://www.w3.org/TR/rdf-canon/#ca.5.3) and related hash collisions in [step 5 in Hash N-Degree Quads](https://www.w3.org/TR/rdf-canon/#hndq.5).

Reply:
Do you have a suggestion to reword this that better reflects your concerns while still showing the robustness of SHA-256?

Reply:
```suggestion
      values are revealed, and the algorithms are designed to cope
      with first-degree hash collisions.</p>
```
By focusing on areas we can address clearly, I believe it becomes more accurate. How about the above rewording?